src/drivers/gps/params.c

```diff
@@ -86,3 +107,42 @@ PARAM_DEFINE_INT32(GPS_UBX_DYNMODEL, 7);
  */
 PARAM_DEFINE_FLOAT(GPS_YAW_OFFSET, 0.f);
 
+/**
+ * Protocol for Main GPS
+ *
+ * Select the GPS protocol over serial.
+ *
+ * Auto-detection will probe all protocols, and thus is a bit slower.
+ *
+ * @min 0
+ * @max 4
+ * @value 0 Auto detect
+ * @value 1 u-blox
+ * @value 2 MTK
+ * @value 3 Ashtech / Trimble
+ * @value 4 Emlid Reach
+ *
+ * @reboot_required true
+ * @group GPS
+ */
+PARAM_DEFINE_INT32(GPS_1_PROTOCOL, 1);
```

Comment:
I think default value should be "Auto detect"
If someone when a faster detection, he can set the protocol.

Reply:
I don't mind too much, but the vast majority of users are using u-blox and therefore automatically profit w/o having to know about the parameter.